lib/json_schemer/schema/base.rb

```diff
@@ -615,13 +626,19 @@ def resolve_ids(schema, ids = {}, parent_uri = nil, pointer = '')
         if schema.is_a?(Array)
           schema.each_with_index { |subschema, index| resolve_ids(subschema, ids, parent_uri, "#{pointer}/#{index}") }
         elsif schema.is_a?(Hash)
-          id = schema[id_keyword]
-          uri = join_uri(parent_uri, id)
-          unless uri == parent_uri
-            ids[uri.to_s] = {
-              schema: schema,
-              pointer: pointer
-            }
+          uri = join_uri(parent_uri, schema[id_keyword])
+          schema.each_pair do |key, value| 
+            if key == id_keyword
+              unless uri == parent_uri
+                ids[uri.to_s] = {
+                  schema: schema,
+                  pointer: pointer
+                }
+              end
+            end
+            if value.is_a?(Hash) 
+              resolve_ids(value, ids, parent_uri, "#{pointer}/#{key}") 
```

Comment:
This should be called on every value, not just Hash values.  (Specifically `Array` is important, and a separate test for it would be nice.)